databricks/koalas/mlflow.py

```diff
@@ -0,0 +1,166 @@
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+
+"""
+MLflow-related functions to load models and apply them to Koalas dataframes.
+"""
+from mlflow.pyfunc import PythonModel, load_pyfunc, spark_udf
+from pyspark.sql.types import DataType
+import pandas as pd
+import numpy as np
+
+from databricks.koalas.utils import lazy_property, default_session
+from databricks.koalas import Series, DataFrame
+from databricks.koalas.typedef import as_spark_type
+
+__all__ = ["PythonModelWrapper", "load_model"]
+
+
+class PythonModelWrapper(object):
+    """
+    A wrapper around MLflow's Python object model.
+
+    This wrapper acts as a predictor on koalas
+
+    """
+    def __init__(self, path, run_id, return_type_hint):
+        self._path = path  # type: str
+        self._run_id = run_id  # type: str
+        self._return_type_hint = return_type_hint
+
+    @lazy_property
+    def _return_type(self) -> DataType:
+        hint = self._return_type_hint
+        # The logic is simple for now, because it corresponds to the default
+        # case: continuous predictions
+        # TODO: do something smarter, for example when there is a sklearn.Classifier (it should
+        # return an integer or a categorical)
+        # We can do the same for pytorch/tensorflow/keras models by looking at the output types.
+        # However, this is probably better done in mlflow than here.
+        if hint == 'infer' or not hint:
+            hint = np.float64
+        return as_spark_type(hint)
+
+    @lazy_property
+    def _model(self) -> PythonModel:
+        return load_pyfunc(self._path, self._run_id)
+
+    @lazy_property
+    def _model_udf(self):
+        spark = default_session()
+        return spark_udf(spark, self._path, self._run_id, result_type=self._return_type)
+
+    def __str__(self):
+        return "PythonModelWrapper({})".format(str(self._model))
+
+    def __repr__(self):
+        return "PythonModelWrapper({})".format(repr(self._model))
+
+    def predict(self, data):
+        """
+        Returns a prediction on the data.
+        If the data is a pandas Dataframe, the return is a pandas series (TODO: np.array?)
+        If the data is a koalas DataFrame, the return is a Koalas Series.
+        """
+        if isinstance(data, pd.DataFrame):
+            return self._model.predict(data)
+        if isinstance(data, DataFrame):
+            cols = [data._sdf[n] for n in data.columns]
+            return_col = self._model_udf(*cols)
+            return Series(data=return_col, anchor=data, index=data._metadata.index_map)
+
+
+def load_model(path, run_id=None, predict_type='infer') -> PythonModelWrapper:
+    """
+    Loads an MLflow model into an wrapper that can be used both for pandas and Koalas dataframes.
+
+    Parameters
+    ----------
+    path : the path of the model, as logged when calling mlflow.log_model or mlflow.save_model
```

Comment:
i think the docs convention is to just explain the data type and default/optional value in the first line, and leave the actual explanation of the parameter on the 2nd line. they are rendered differently.